### PR TITLE
generate: Add possibility to install additional packages in image

### PR DESCRIPTION
**Pull request by creydr**
Adds the `additional-packages` arg, which allows to specify additional packages which should be installed into the image.
This can be helpful for migrating the kn-plugin-func utils [Dockerfile](https://github.com/openshift-knative/kn-plugin-func/blob/serverless-1.33/Dockerfile.utils), which needs additional packages.

### CODE REVIEW

File: cmd/generate/Dockerfile.template [hunk 1]
```diff
@@ -18,6 +18,12 @@ ARG VERSION={{.version}}
 # Install zoneinfo.
 RUN microdnf install tzdata
 
+{{- if .additional_packages}}
+
+# Install additional packages
+RUN microdnf install {{.additional_packages}}
+{{- end}}
```
[the remark this answers is not in the document]
Comment (pierDipi) [reply cmd/generate/Dockerfile.template:25]: and that produces only one layer

[the remark this answers is not in the document]
Comment (pierDipi) [reply cmd/generate/Dockerfile.template:25]: also `if` not needed ?

[the remark this answers is not in the document]
Comment (creydr) [reply cmd/generate/Dockerfile.template:25]: I was doing that before https://github.com/openshift-knative/hack/pull/240/commits/e8d8524492523664219e8ebc1da993443197f881. But then thought it's clearer, as we then can keep the zoneinfo comment

File: pkg/project/testoutput/openshift/ci-operator/knative-images/discover/Dockerfile [hunk 2]
```diff
@@ -16,7 +16,7 @@ FROM $GO_RUNTIME
 ARG VERSION=main
 
 # Install zoneinfo.
```
Comment (pierDipi) [inline pkg/project/testoutput/openshift/ci-operator/knative-images/discover/Dockerfile:18]:
```suggestion
```

**Pull request merged**